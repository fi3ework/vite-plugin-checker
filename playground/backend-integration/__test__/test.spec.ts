import stringify from 'fast-json-stable-stringify'
import { describe, expect, it } from 'vitest'

import {
  diagnostics,
  expectStderrContains,
  isBuild,
  isServe,
  log,
  sleepForServerReady,
  stripedLog,
} from '../../testUtils'

describe('backend-integration', () => {
  describe.runIf(isServe)('serve', () => {
    it('get initial error', async () => {
      await sleepForServerReady()

      expect(stringify(diagnostics)).toMatchSnapshot()
      expect(stripedLog).toMatchSnapshot()
    })
  })

  describe.runIf(isBuild)('build', () => {
    it('should fail', async () => {
      const expectedMsg = `TS2345: Argument of type 'number' is not assignable to parameter of type 'string | (() => string)'`
      expectStderrContains(log, expectedMsg)
    })
  })
})
