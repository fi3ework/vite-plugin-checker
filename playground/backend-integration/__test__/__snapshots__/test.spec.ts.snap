// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`backend-integration > serve > get initial error 1`] = `"[{"checkerId":"TypeScript","frame":"    4 |/n    5 | function App() {/n  > 6 |   const [count, setCount] = useState<string>(0)/n      |                                              ^/n    7 |/n    8 |   return (/n    9 |     <div className=/"App/">","id":"<PROJECT_ROOT>/playground-temp/backend-integration/src/App.tsx","level":1,"loc":{"column":46,"file":"<PROJECT_ROOT>/playground-temp/backend-integration/src/App.tsx","line":6},"message":"Argument of type 'number' is not assignable to parameter of type 'string | (() => string)'.","stack":""}]"`;

exports[`backend-integration > serve > get initial error 2`] = `
"
 ERROR(TypeScript)  Argument of type 'number' is not assignable to parameter of type 'string | (() => string)'.
 FILE  <PROJECT_ROOT>/playground-temp/backend-integration/src/App.tsx:6:46

    4 |
    5 | function App() {
  > 6 |   const [count, setCount] = useState<string>(0)
      |                                              ^
    7 |
    8 |   return (
    9 |     <div className="App">

[TypeScript] Found 1 error. Watching for file changes."
`;
